spec/javascripts/govuk_publishing_components/analytics-ga4/ga4-form-tracker.spec.js

```diff
@@ -0,0 +1,211 @@
+/* eslint-env jasmine */
+
+describe('Google Analytics form tracking', function () {
+  var GOVUK = window.GOVUK
+  var element
+  var expected
+
+  function agreeToCookies () {
+    GOVUK.setCookie('cookies_policy', '{"essential":true,"settings":true,"usage":true,"campaigns":true}')
+  }
+
+  function denyCookies () {
+    GOVUK.setCookie('cookies_policy', '{"essential":false,"settings":false,"usage":false,"campaigns":false}')
+  }
+
+  beforeAll(function () {
+    window.GOVUK.analyticsGa4 = window.GOVUK.analyticsGa4 || {}
+    window.GOVUK.analyticsGa4.vars = window.GOVUK.analyticsGa4.vars || {}
+    window.GOVUK.analyticsGa4.vars.gem_version = 'aVersion'
+  })
+
+  beforeEach(function () {
+    window.dataLayer = []
+    element = document.createElement('form')
+    document.body.appendChild(element)
+    agreeToCookies()
+  })
+
+  afterEach(function () {
+    document.body.removeChild(element)
+  })
+
+  afterAll(function () {
+    window.dataLayer = []
+  })
+
+  describe('when the user has a cookie consent choice', function () {
+    it('starts the module if consent has already been given', function () {
+      agreeToCookies()
+      var tracker = new GOVUK.Modules.Ga4FormTracker(element)
+      spyOn(tracker, 'trackFormSubmit')
```

Comment:
I hadn't realised. Is there a direct alternative?

Reply:
If I remember correctly, replacing `spyOn` turned out to be a bit of a rabbit hole and could potentially hold up this work from being merged whilst a vanilla JS alternative is found. Do you think we could use `spyOn` for now with the proviso that it will be replaced later?

Reply:
Works for me.